test/libsolidity/util/TestFunctionCall.h

```diff
@@ -0,0 +1,88 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <test/libsolidity/util/TestFileParser.h>
+#include <test/TestCase.h>
+
+#include <libsolidity/ast/Types.h>
+#include <liblangutil/Exceptions.h>
+#include <libdevcore/CommonData.h>
+
+#include <iosfwd>
+#include <numeric>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+/**
+ * Represents a function call and the result it returned. It stores the call
+ * representation itself, the actual byte result (if any) and a string representation
+ * used for the interactive update routine provided by isoltest. It also provides
+ * functionality to compare the actual result with the expectations attached to the
+ * call object, as well as a way to reset the result if executed multiple times.
+ */
+class TestFunctionCall
+{
+public:
+	using FunctionCall = dev::solidity::test::FunctionCall;
+	using ParamaterList = dev::solidity::test::ParameterList;
+
+	TestFunctionCall(FunctionCall _call): m_call(_call) {}
```

Comment:
non-explicit on purpose?

Reply:
Also pass-by-value but no move?

Reply:
Ah, should move, yes. Now, I need the constructor to be non-explicit, because `std::back_inserter` can't construct a `TestFunctionCall` out of a `FunctionCall` otherwise in `SemanticTest::parseExpectations()`.